src/components/ChannelPreview/ChannelPreviewMessenger.js

```diff
@@ -63,104 +64,68 @@ const Message = styled.Text`
-     * are displayed in 'DD/MM/YY' format. props.latestMessage.created_at is the
-     * default formated date. This default logic is part of ChannelPreview component.
-     */
-    formatLatestMessageDate: PropTypes.func,
-  };
-
-  static defaultProps = {
-    latestMessageLength: 30,
-  };
-
-  onSelectChannel = () => {
-    this.props.setActiveChannel(this.props.channel);
-  };
-
-  renderAvatar = (otherMembers) => {
-    const { channel } = this.props;
-    if (channel.data.image)
-      return (
-        <Avatar image={channel.data.image} size={40} name={channel.data.name} />
-      );
-
-    if (otherMembers.length === 1)
-      return (
-        <Avatar
-          image={otherMembers[0].user.image}
-          size={40}
-          name={channel.data.name || otherMembers[0].user.name}
-        />
-      );
-
-    return <Avatar size={40} name={channel.data.name} />;
-  };
-
-  render() {
-    const { channel, t } = this.props;
-    let otherMembers = [];
-    let name = channel.data.name;
-    const isValidName = name && typeof name === 'string';
-
-    if (!isValidName) {
-      const members = channel.state ? Object.values(channel.state.members) : [];
-      otherMembers = members.filter(
-        (member) => member.user.id !== this.props.client.userID,
-      );
-      name = otherMembers
-        .map((member) => member.user.name || member.user.id || 'Unnamed User')
-        .join(', ');
-    }
-    const formatLatestMessageDate = this.props.formatLatestMessageDate;
-    return (
-      <Container onPress={this.onSelectChannel}>
-        {this.renderAvatar(otherMembers)}
-        <Details>
-          <DetailsTop>
-            <Title ellipsizeMode="tail" numberOfLines={1}>
-              {name}
-            </Title>
-            <Date>
-              {formatLatestMessageDate
-                ? formatLatestMessageDate(
-                    this.props.latestMessage.messageObject.created_at,
-                  )
-                : this.props.latestMessage.created_at}
-            </Date>
-          </DetailsTop>
-          <Message
-            unread={this.props.unread > 0 ? this.props.unread : undefined}
-          >
-            {!this.props.latestMessage
-              ? t('Nothing yet...')
-              : truncate(this.props.latestMessage.text.replace(/\n/g, ' '), {
-                  length: this.props.latestMessageLength,
-                })}
-          </Message>
-        </Details>
-      </Container>
-    );
-  }
-}
-
-export default withTranslationContext(themed(ChannelPreviewMessenger));
+const ChannelPreviewMessenger = ({
+  channel,
+  latestMessage,
+  unread,
+  latestMessageLength = 30,
+  formatLatestMessageDate,
+  setActiveChannel,
+}) => {
+  const displayName = useChannelPreviewDisplayName(channel);
+  const displayAvatar = useChannelPreviewDisplayAvatar(channel);
+
+  return (
+    <Container
+      onPress={setActiveChannel.bind(null, channel)}
+      testID="channel-preview-button"
```

Comment:
we should be consistent with single vs double quotes? I think single is better cause it's 1 less key stroke cause you don't need to use the `shift` key but idc too much which we choose
```suggestion
      testID='channel-preview-button'
```